lib/new_relic/agent/instrumentation/elasticsearch/instrumentation.rb

```diff
@@ -56,7 +56,11 @@ def nr_cluster_name
       return if nr_hosts.empty?
 
       NewRelic::Agent.disable_all_tracing do
-        @nr_cluster_name ||= perform_request('GET', '_cluster/health').body['cluster_name']
+        @nr_cluster_name ||= if NewRelic::Agent.config[:'elasticsearch.perform_health_check']
+          perform_request('GET', '_cluster/health').body['cluster_name']
+        else
+          NewRelic::UNKNOWN
```

Comment:
Technically the `database_name` attribute on a datastore segment can be nil. Would that make more sense here than `Unknown`? I think it's closer to the API's expectations.

Reply:
Hmm I think one of the benefits to not using nil is that we won't have to check the config every single time. If we did set it to nil, then line 55 wouldn't return early anymore, so we'd have to do everything else every time and itd take a bit longer. Not sure if it would be a concerning increase though, or not.

Reply:
That's right! Thanks for setting me straight :) 

We could add another guard clause between lines 54 and 55 to check for the config and return early if it's false, but that might still be less performant than just setting "Unknown".

Reply:
I think `'Unknown'` vs `nil` is worth discussing for what gets sent up in the db span. Originally I felt that `'Unknown'` would convey a failure to obtain the name or a deliberate short-circuit preventing the discovery, whereas `nil` would convey that the name is unset (given that Rails apps don't require the name in `database.yml`).

If `nil` is thought to be more appropriate, we can work around the performance implications. Perhaps `defined?` is faster than a hash lookup.

```diff
- @nr_cluster_name ||= if NewRelic::Agent.config[:'elasticsearch.perform_health_check']
+ return @nr_cluster_name if defined?(@nr_cluster_name)
+
+ @nr_cluster_name = if NewRelic::Agent.config[:'elasticsearch.perform_health_check']
```

Reply:
`start_datastore_segment`'s default arguments set `database_name` to `nil`, which leads me to believe that's a more standard value than `Unknown`. 

The code change I alluded to slightly differs from what you've suggested above.
Using the method as it exists on `dev`, the change would be:
```diff
    def nr_cluster_name
+    return unless NewRelic::Agent.config[:'elasticsearch.perform_health_check']
      return @nr_cluster_name if @nr_cluster_name
      return if nr_hosts.empty?

      NewRelic::Agent.disable_all_tracing do
        @nr_cluster_name ||= perform_request('GET', '_cluster/health').body['cluster_name']
      end
    rescue StandardError => e
      NewRelic::Agent.logger.error('Failed to get cluster name for elasticsearch', e)
      nil
    end
```

So the `NewRelic::Agent.config` check would occur every time `#nr_cluster_name` is called, instead of when the `@nr_cluster_name` variable is defined/assigned. I don't know if there's a way to make that config check more performant.

Reply:
Ah, I was definitely thinking from the perspective of a change to config option only taking effect on a complete restart of the agent, so we'd only check the config hash one time and then either cache the string cluster name value or `nil`. And then use `defined?` for memoization help with `nil`. If you actually want to check the config freshly each time, then a hash lookup in a relatively small hash with symbols as keys shouldn't be too bad and the code as-is works great.

Reply:
Ohh, I understand now. Thanks for clarifying! Your latest comment made me realize we don't need to check the config that often. It's reasonable to expect the config won't change during Elasticsearch client initialization. I was getting overly fixated on the test scenario where I needed to create a new client. Thank you!